Expensify/App
# Setup "Connect bank account" workspace flow

### Details
There's a lot of cleanup that needs to happen here (translation / style issues) + still waiting for assets but mostly done.

### Fixed Issues
$ https://github.com/Expensify/App/issues/5697
$ https://github.com/Expensify/App/issues/5168
$ https://github.com/Expensify/App/issues/5707

### Tests
1. Sign out of both New and Old Expensify
1. Create a new account + workspace
1. Navigate to the workspace 
1. Tap "Connect bank account"
1. Verify you see this view
![2021-10-07_07-02-59](https://user-images.githubusercontent.com/32969087/136431062-a001c895-d4c7-4076-a92f-e3d17de1dc04.png)
1. Select "Connect manually"
1. Close the view by pressing the `x` here
![2021-10-06_13-49-11](https://user-images.githubusercontent.com/32969087/136298831-9ae80822-74d5-45c6-8063-59aaad0bbe4f.png)
1. Navigate to the workspace again
1. Tap "Connect bank account"
1. Verify you land on the "Connect manually" screen
1. Verify the account add method memory also works for the "Connect online with Plaid" option
1. Connect with Plaid using the test credentials `bank: chase, user: user_good, pass: pass_good`
1. Select Plaid savings account from the drop down
1. Begin filling out the form so that you end up with an [OPEN bank account](https://stackoverflow.com/c/expensify/questions/342)
1. Stop after the "Company Information" step and close out the flow
1. Reopen and verify you see this screen 
![2021-10-07_07-01-54](https://user-images.githubusercontent.com/32969087/136431126-2cde2b64-1555-461e-a163-a4e2832fc4fe.png)
1. Press "Continue with setup" and verify you are brought back to the next step you need to complete ("Personal Information")
1. Continue forward and complete the "Personal Information" step and exit the flow again
1. Reopen the flow again and press "Continue with setup" - verify you are brought back to the next step you need to complete ("Additional Information")
1. Fill out the "Additional information" step and continue 
1. Verify you see this screen
![2021-10-07_07-25-06](https://user-images.githubusercontent.com/32969087/136433921-e98eb3ca-fe22-4ed7-9c47-7c8940f7ca53.png)
1. Verify that tapping "Chat with Concierge" opens up the chat with Concierge
1. Add a secondary login with a non public domain + enable Expensify cards
1. Return to the "Connect bank account" flow
1. Verify you now see this view
![2021-10-07_07-40-51](https://user-images.githubusercontent.com/32969087/136435986-1a0d4e34-84ce-4a81-a99e-7870e2fc8d2c.png)

**Repeat flow with new account and test `VERIFYING` account**
1. Complete the flow and stop after the company info step
2. Close the modal
3. Reopen and verify "Continue with setup" screen appears
4. Verify you are brought back to the "Personal Information" step
5. Fill out this step
6. Verify Onfido step appears and close the modal
7. Verify that when the flow is reopened we are still on the "Personal Information" step
8. Pass Onfido flow by entering information and submitting
9. Once on the "Additional Information" step close out of the flow
10. Reopen and verify you are brought back to the "Additional Information" step
11. Go through this step and verify you see this screen
![2021-10-07_07-58-17](https://user-images.githubusercontent.com/32969087/136438375-55b064b7-c42d-45c1-b800-f32c75aee516.png)
12. Tap "Yes, let's chat" and verify you are brought to the Concierge chat
13. Have ops verify this account and get the 3 validation amounts to input
14. Return to the flow again and enter the 3 amounts
15. Verify you end up on this screen (or possibly the variant with cards? I won't be trying to get the cards provisioned on dev rn sorry - if this was with a non public domain it should have auto-provisioned)
![2021-10-07_08-05-04](https://user-images.githubusercontent.com/32969087/136439209-51613e07-189f-4a0d-a55d-7d832e21c3cc.png)

### QA Steps
Internal QA

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
